plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationProviderImpl.kt

```diff
@@ -26,11 +29,26 @@ internal class LocationProviderImpl(context: Context) : LocationProvider, Locati
 
   private val locationConsumers = CopyOnWriteArrayList<LocationConsumer>()
 
+  private var handler: Handler? = null
+  private var runnable: Runnable? = null
+
   @SuppressLint("MissingPermission")
   private fun requestLocationUpdates() {
-    locationEngine.requestLocationUpdates(
-      locationEngineRequest, this, Looper.getMainLooper()
-    )
+    if (PermissionsManager.areLocationPermissionsGranted(context)) {
+      locationEngine.requestLocationUpdates(
+        locationEngineRequest, this, Looper.getMainLooper()
+      )
+    } else {
+      if (handler == null) {
+        handler = Handler()
+        runnable = Runnable { requestLocationUpdates() }
+      }
+      handler?.postDelayed(runnable, UPDATE_DELAY)
+      Logger.w(
+        TAG,
+        "Missing location permission, location component will not take effect before location permission is granted."
```

Comment:
I'm ok with a loop that verifies if permissions have been expected but 2 remarks:
 - shouldn't we clean up the handler when the location/map is destroyed (we could end up with a loop that continuous forever)
 - can we apply an exponential backoff + decrease the initial UPDATE_DELAY so it eventually grows to the current value?

Reply:
The runnable will be removed in `unRegisterLocationConsumer` which would be invoked while deactivateLocationComponent